app/src/main/java/org/dhis2/usescases/teiDashboard/DashboardRepositoryImpl.kt

```diff
@@ -75,23 +80,27 @@ class DashboardRepositoryImpl(
         return d2.eventModule().events().byEnrollmentUid().eq(enrollmentUid)
             .byDeleted().isFalse
             .orderByTimeline(RepositoryScope.OrderByDirection.ASC)
-            .get().toFlowable().flatMapIterable { events: List<Event>? -> events }.distinct()
-            .map { event: Event ->
-                var event = event
-                if (java.lang.Boolean.FALSE
-                    == d2.programModule().programs().uid(programUid).blockingGet()!!
-                        .ignoreOverdueEvents()
-                ) if (event.status() == EventStatus.SCHEDULE &&
-                    event.dueDate()!!
-                        .before(DateUtils.getInstance().today)
-                ) {
-                    event = updateState(event, EventStatus.OVERDUE)
+            .get()
+            .map { events ->
+                events.map { event ->
+                    if (shouldUpdateStatusToOverdue(event)) {
+                        updateState(event, EventStatus.OVERDUE)
+                    } else {
+                        event
+                    }
                 }
-                event
-            }.toList()
+            }
             .toObservable()
     }
 
+    private fun shouldUpdateStatusToOverdue(event: Event): Boolean {
+        val ignoreOverdueEvents = d2.program(programUid!!)?.ignoreOverdueEvents()
+        val scheduledEventIsOverdue = event.status() == EventStatus.SCHEDULE &&
+            event.dueDate()!!.before(DateUtils.getInstance().today)
```

Comment:
I know that this pr just adds a breadcrum. However, I think this will crash if programUid is null, like when entering the all enrollments dashboard.